_posts/2025-09-30-add-new-table-to-capture-list.adoc

```diff
@@ -0,0 +1,143 @@
+
+Metadata messages:: PostgreSQL uses https://www.postgresql.org/docs/current/logicaldecoding-explanation.html[Logical
+ Decoding] to send relation (table) metadata messages.
+Debezium receives a relation message (https://www.postgresql.org/docs/current/protocol-logicalrep-message-formats.html#PROTOCOL-LOGICALREP-MESSAGE-FORMATS-RELATION[`R`]) before the first change event for a table is sent, whenever a schema change occurs, or when replication resumes.
+These messages allow Debezium to know which table a change belongs to, the column order, data types, and other structural information.
+
+JDBC queries:: For connectors such as SQL Server and Db2, if a change event arrives for a table that is not yet in the in-memory schema representation,
+Debezium directly queries the database to retrieve the table schema.
+
+For connectors that do not rely on metadata messages, the schema information is also persisted so that it can survive a connector restart or failure.
+For this purpose, there are several properties prefixed with `schema.history.internal.` that manage configuration in this area.
+Of particular interest this the property:
+
+`schema.history.internal.store.only.captured.tables.ddl`:: Specifies whether the connector records schema structures from all tables in a schema or database, or only from tables that are designated for capture.
+
+There is also a related property:
+
+`schema.history.internal.store.only.captured.databases.ddl`:: Specifies whether the connector records schema structures from all databases or only from databases that are designated for capture.
+
+For now remember it, we will see in detail what role it has later.
+
+Now, let’s take a step back: what happens when a connector is started for the first time?
+
+Before emitting events for a table, Debezium needs to know the table structure.
+Even though this information could be obtained from a DDL change while streaming, it’s unlikely that the DDL is still present in the logs when the connector starts for the very first time.
+This is why Debezium requires capturing the database schema at first start-up.
+
+As a side note, the initial snapshot exists primarily to capture the current data present in the database,
+since not all data may be available in the transaction log when Debezium starts for the very first time.
+Capturing the schema is required so that Debezium knows the table structures.
+Even if you don’t want to capture the existing data, schema capture cannot be disabled.
+You can, however, skip capturing the data itself by using the no_data snapshot mode.
+
+That said, you should start to understand why the answer is “it depends.”
+
+Let’s walk through a couple of common scenarios you may run into.
+
+=== Scenario 1
+
+Let’s start with an easy scenario: you’ve set the `*.include.list` properties and ended up capturing all available tables in the database.
```

Comment:
I think you may have intended this to read "you've not set the `*.include.list` properties" ?

Reply:
Well the intention here is you have set the combination of `database.include.list`, `schema.include.list`, and `table.include.list` so that you effectively capture all tables. Dunno if it's clear...

Reply:
I see, then would it make more sense to mention by either leaving these 3 configurations unset or by expressly setting these to include all tables, since both of those combinations yield the same net result?